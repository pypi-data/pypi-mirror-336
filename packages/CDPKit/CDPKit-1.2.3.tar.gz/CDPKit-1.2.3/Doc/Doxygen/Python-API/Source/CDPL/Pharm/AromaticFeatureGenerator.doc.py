#
# This file is part of the Chemical Data Processing Toolkit
#
# Copyright (C) Thomas Seidel <thomas.seidel@univie.ac.at>
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU Lesser General Public
# License as published by the Free Software Foundation; either
# version 2 of the License, or (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
# Lesser General Public License for more details.
#
# You should have received a copy of the GNU Lesser General Public License
# along with this program; see the file COPYING. If not, write to
# the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
# Boston, MA 02111-1307, USA.
#

##
# \brief AromaticFeatureGenerator.
# 
class AromaticFeatureGenerator(PatternBasedFeatureGenerator):

    ##
    # \brief 
    #
    DEF_FEATURE_TOL = 0.9

    ##
    # \brief 
    #
    DEF_FEATURE_TYPE = 2

    ##
    # \brief 
    #
    DEF_FEATURE_GEOM = 3

    ##
    # \brief Constructs the <tt>AromaticFeatureGenerator</tt> instance.
    # 
    def __init__() -> None: pass

    ##
    # \brief Constructs a copy of the <tt>AromaticFeatureGenerator</tt> instance <em>gen</em>.
    # 
    # \param gen The <tt>AromaticFeatureGenerator</tt> instance to copy.
    # 
    def __init__(gen: AromaticFeatureGenerator) -> None: pass

    ##
    # \brief Perceives aromatic ring features of the molecular graph a\ molgraph and adds them to the pharmacophore <em>pharm</em>.
    # 
    # \param molgraph The molecular graph for which to perceive the features.
    # \param pharm The output pharmacophore where to add the generated features.
    # 
    def __init__(molgraph: Chem.MolecularGraph, pharm: Pharmacophore) -> None: pass

    ##
    # \brief Specifies the value of the feature type property that has to be set on newly generated features.
    # 
    # \param type The value of the feature type property.
    # 
    # \note The default type is specified by the constant AromaticFeatureGenerator.DEF_FEATURE_TYPE. 
    # 
    # \see FeatureProperty.TYPE
    # 
    def setFeatureType(type: int) -> None: pass

    ##
    # \brief Specifies the value of the feature geometry property that has to be set on newly generated features.
    # 
    # \param geom The value of the feature geometry property.
    # 
    # \note The default type is specified by the constant AromaticFeatureGenerator.DEF_FEATURE_GEOM. 
    # 
    # \see FeatureProperty.GEOMETRY
    # 
    def setFeatureGeometry(geom: int) -> None: pass

    ##
    # \brief Specifies the value of the feature tolerance property that has to be set on newly generated features.
    # 
    # \param tol The value of the feature tolerance property.
    # 
    # \note The default value is specified by the constant AromaticFeatureGenerator.DEF_FEATURE_TOL. 
    # 
    # \see FeatureProperty.TOLERANCE
    # 
    def setFeatureTolerance(tol: float) -> None: pass

    ##
    # \brief Returns the value of the feature type property that gets set on newly generated features.
    # 
    # \return The used value of the feature type property. 
    # 
    # \see FeatureProperty.TYPE
    # 
    def getFeatureType() -> int: pass

    ##
    # \brief Returns the value of the feature geometry property that gets set on newly generated features.
    # 
    # \return The used value of the feature geometry property. 
    # 
    # \see FeatureProperty.GEOMETRY
    # 
    def getFeatureGeometry() -> int: pass

    ##
    # \brief Returns the value of the feature tolerance property that gets set on newly generated features.
    # 
    # \return The value of the feature tolerance property. 
    # 
    # \see FeatureProperty.TOLERANCE
    # 
    def getFeatureTolerance() -> float: pass

    ##
    # \brief Replaces the current set include/exclude patterns and attributes by a copy of the <tt>AromaticFeatureGenerator</tt> instance <em>gen</em>.
    # 
    # \param gen The <tt>AromaticFeatureGenerator</tt> instance to copy.
    # 
    # \return \a self
    # 
    def assign(gen: AromaticFeatureGenerator) -> AromaticFeatureGenerator: pass

    featureType = property(getFeatureType, setFeatureType)

    featureGeometry = property(getFeatureGeometry, setFeatureGeometry)

    featureTolerance = property(getFeatureTolerance, setFeatureTolerance)
