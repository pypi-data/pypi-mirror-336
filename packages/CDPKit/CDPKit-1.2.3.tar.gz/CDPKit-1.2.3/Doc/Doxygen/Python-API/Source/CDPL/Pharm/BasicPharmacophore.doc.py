#
# This file is part of the Chemical Data Processing Toolkit
#
# Copyright (C) Thomas Seidel <thomas.seidel@univie.ac.at>
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU Lesser General Public
# License as published by the Free Software Foundation; either
# version 2 of the License, or (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
# Lesser General Public License for more details.
#
# You should have received a copy of the GNU Lesser General Public License
# along with this program; see the file COPYING. If not, write to
# the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
# Boston, MA 02111-1307, USA.
#

##
# \brief BasicPharmacophore.
# 
class BasicPharmacophore(Pharmacophore):

    ##
    # \brief Constructs an empty <tt>BasicPharmacophore</tt> instance.
    # 
    def __init__() -> None: pass

    ##
    # \brief Constructs a copy of the Pharm.Pharmacophore instance <em>pharm</em>.
    # 
    # \param pharm The other Pharm.Pharmacophore instance to copy.
    # 
    def __init__(pharm: BasicPharmacophore) -> None: pass

    ##
    # \brief Constructs a copy of the Pharm.Pharmacophore instance <em>pharm</em>.
    # 
    # \param pharm The other Pharm.Pharmacophore instance to copy.
    # 
    def __init__(pharm: Pharmacophore) -> None: pass

    ##
    # \brief Constructs a BasicPharmacophore instance with copies of the features in the Pharm.FeatureContainer instance <em>cntnr</em>.
    # 
    # \param cntnr The Pharm.FeatureContainer instance with the features to copy.
    # 
    def __init__(cntnr: FeatureContainer) -> None: pass

    ##
    # \brief Replaces the current set of pharmacophore features and properties by a copy of the features and properties of the pharmacophore <em>pharm</em>.
    # 
    # \param pharm The pharmacophore to copy.
    # 
    def copy(pharm: BasicPharmacophore) -> None: pass

    ##
    # \brief Replaces the current set of pharmacophore features and properties by a copy of the features and properties of the pharmacophore <em>pharm</em>.
    # 
    # \param pharm The pharmacophore to copy.
    # 
    def copy(pharm: Pharmacophore) -> None: pass

    ##
    # \brief Replaces the current set of pharmacophore features and properties by a copy of the features and properties of the feature container <em>cntnr</em>.
    # 
    # \param cntnr The Pharm.FeatureContainer instance providing the features and properties to copy.
    # 
    def copy(cntnr: FeatureContainer) -> None: pass

    ##
    # \brief Extends the current set of pharmacophore features by a copy of the features in the pharmacophore <em>pharm</em>.
    # 
    # \param pharm The pharmacophore providing the features to append.
    # 
    # \note Does not affect any properties.
    # 
    def append(pharm: BasicPharmacophore) -> None: pass

    ##
    # \brief Extends the current set of pharmacophore features by a copy of the features in the pharmacophore <em>pharm</em>.
    # 
    # \param pharm The pharmacophore providing the features to append.
    # 
    # \note Does not affect any properties.
    # 
    def append(pharm: Pharmacophore) -> None: pass

    ##
    # \brief Extends the current set of pharmacophore features by a copy of the features in the feature container <em>cntnr</em>.
    # 
    # \param cntnr The Pharm.FeatureContainer instance providing the features to append.
    # 
    # \note Does not affect any properties.
    # 
    def append(cntnr: FeatureContainer) -> None: pass

    ##
    # \brief Replaces the current set of pharmacophore features and properties by a copy of the features and properties of the pharmacophore <em>pharm</em>.
    # 
    # Internally calls copy() to perform the actual work.
    # 
    # \param pharm The pharmacophore to copy.
    # 
    # \return \a self
    # 
    def assign(pharm: BasicPharmacophore) -> BasicPharmacophore: pass

    ##
    # \brief Replaces the current set of pharmacophore features and properties by a copy of the features and properties of the pharmacophore <em>pharm</em>.
    # 
    # Internally calls copy() to perform the actual work.
    # 
    # \param pharm The pharmacophore to copy.
    # 
    # \return \a self
    # 
    def assign(pharm: Pharmacophore) -> BasicPharmacophore: pass

    ##
    # \brief Replaces the current set of pharmacophore features and properties by a copy of the features and properties of the feature container <em>cntnr</em>.
    # 
    # Internally calls copy() to perform the actual work.
    # 
    # \param cntnr The Pharm.FeatureContainer instance providing the features and properties to copy.
    # 
    # \return \a self
    # 
    def assign(cntnr: FeatureContainer) -> BasicPharmacophore: pass

    ##
    # \brief Performs the in-place addition operation <tt>self += pharm</tt>.
    # \param pharm Specifies the second addend.
    # \return The updated \c %BasicPharmacophore instance \a self.
    # 
    def __iadd__(pharm: BasicPharmacophore) -> BasicPharmacophore: pass

    ##
    # \brief Performs the in-place addition operation <tt>self += pharm</tt>.
    # \param pharm Specifies the second addend.
    # \return The updated \c %BasicPharmacophore instance \a self.
    # 
    def __iadd__(pharm: Pharmacophore) -> BasicPharmacophore: pass

    ##
    # \brief Performs the in-place addition operation <tt>self += cntnr</tt>.
    # \param cntnr Specifies the second addend.
    # \return The updated \c %BasicPharmacophore instance \a self.
    # 
    def __iadd__(cntnr: FeatureContainer) -> BasicPharmacophore: pass
