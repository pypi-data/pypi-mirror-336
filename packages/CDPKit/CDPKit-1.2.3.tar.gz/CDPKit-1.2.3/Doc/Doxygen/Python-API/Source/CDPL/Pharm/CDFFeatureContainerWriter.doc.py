#
# This file is part of the Chemical Data Processing Toolkit
#
# Copyright (C) Thomas Seidel <thomas.seidel@univie.ac.at>
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU Lesser General Public
# License as published by the Free Software Foundation; either
# version 2 of the License, or (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
# Lesser General Public License for more details.
#
# You should have received a copy of the GNU Lesser General Public License
# along with this program; see the file COPYING. If not, write to
# the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
# Boston, MA 02111-1307, USA.
#

##
# \brief A writer for feature container data in the native I/O format of the <em>CDPL</em>.
# 
class CDFFeatureContainerWriter(FeatureContainerWriterBase):

    ##
    # \brief Constructs a <tt>CDFFeatureContainerWriter</tt> instance that will write data of feature containers to the output stream <em>os</em>.
    # 
    # \param os The output stream to write to.
    # 
    def __init__(os: Base.OStream) -> None: pass
