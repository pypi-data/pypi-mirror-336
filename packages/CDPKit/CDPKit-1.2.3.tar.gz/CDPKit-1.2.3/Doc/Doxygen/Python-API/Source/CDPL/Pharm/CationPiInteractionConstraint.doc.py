#
# This file is part of the Chemical Data Processing Toolkit
#
# Copyright (C) Thomas Seidel <thomas.seidel@univie.ac.at>
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU Lesser General Public
# License as published by the Free Software Foundation; either
# version 2 of the License, or (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
# Lesser General Public License for more details.
#
# You should have received a copy of the GNU Lesser General Public License
# along with this program; see the file COPYING. If not, write to
# the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
# Boston, MA 02111-1307, USA.
#

##
# \brief CationPiInteractionConstraint.
# 
class CationPiInteractionConstraint(Boost.Python.instance):

    ##
    # \brief 
    #
    DEF_MIN_DISTANCE = 3.5

    ##
    # \brief 
    #
    DEF_MAX_DISTANCE = 5.5

    ##
    # \brief 
    #
    DEF_MAX_ANGLE = 45.0

    ##
    # \brief Initializes a copy of the \e %CationPiInteractionConstraint instance \a constr.
    # \param constr The \e %CationPiInteractionConstraint instance to copy.
    # 
    def __init__(constr: CationPiInteractionConstraint) -> None: pass

    ##
    # \brief Constructs a <tt>CationPiInteractionConstraint</tt> functor with the specified constraints.
    # 
    # \param aro_cat <tt>True</tt> if the first feature argument represents the aromatic- and the second one the cationic-feature, and <tt>False</tt> otherwise.
    # \param min_dist The minimum allowed distance between the cationic- and aromatic-feature centers.
    # \param max_dist The maximum allowed distance between the cationic- and aromatic-feature centers.
    # \param max_ang The maximum allowed angle between the cationic- and aromatic-feature plane normal.
    # 
    def __init__(aro_cat: bool, min_dist: float = 3.5, max_dist: float = 5.5, max_ang: float = 45.0) -> None: pass

    ##
    # \brief Returns the numeric identifier (ID) of the wrapped C++ class instance.
    # 
    # Different Python \c %CationPiInteractionConstraint instances may reference the same underlying C++ class instance. The commonly used Python expression
    # <tt>a is not b</tt> thus cannot tell reliably whether the two \c %CationPiInteractionConstraint instances \e a and \e b reference different C++ objects. 
    # The numeric identifier returned by this method allows to correctly implement such an identity test via the simple expression
    # <tt>a.getObjectID() != b.getObjectID()</tt>.
    # 
    # \return The numeric ID of the internally referenced C++ class instance.
    # 
    def getObjectID() -> int: pass

    ##
    # \brief 
    # \return 
    #
    def getMinDistance() -> float: pass

    ##
    # \brief 
    # \return 
    #
    def getMaxDistance() -> float: pass

    ##
    # \brief 
    # \return 
    #
    def getMaxAngle() -> float: pass

    ##
    # \brief Replaces the current state of \a self with a copy of the state of the \c %CationPiInteractionConstraint instance \a constr.
    # \param constr The \c %CationPiInteractionConstraint instance to copy.
    # \return \a self
    # 
    def assign(constr: CationPiInteractionConstraint) -> CationPiInteractionConstraint: pass

    ##
    # \brief 
    # \param ftr1 
    # \param ftr2 
    # \return 
    #
    def __call__(ftr1: Feature, ftr2: Feature) -> bool: pass

    objectID = property(getObjectID)

    minDistance = property(getMinDistance)

    maxDistance = property(getMaxDistance)

    maxAngle = property(getMaxAngle)
