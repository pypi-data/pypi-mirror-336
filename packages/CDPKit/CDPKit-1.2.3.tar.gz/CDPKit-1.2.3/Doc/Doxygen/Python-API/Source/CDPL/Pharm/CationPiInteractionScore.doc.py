#
# This file is part of the Chemical Data Processing Toolkit
#
# Copyright (C) Thomas Seidel <thomas.seidel@univie.ac.at>
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU Lesser General Public
# License as published by the Free Software Foundation; either
# version 2 of the License, or (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
# Lesser General Public License for more details.
#
# You should have received a copy of the GNU Lesser General Public License
# along with this program; see the file COPYING. If not, write to
# the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
# Boston, MA 02111-1307, USA.
#

##
# \brief CationPiInteractionScore.
# 
class CationPiInteractionScore(FeatureInteractionScore):

    ##
    # \brief 
    #
    DEF_MIN_DISTANCE = 3.5

    ##
    # \brief 
    #
    DEF_MAX_DISTANCE = 5.5

    ##
    # \brief 
    #
    DEF_MAX_ANGLE = 30.0

    ##
    # \brief Initializes a copy of the \e %CationPiInteractionScore instance \a score.
    # \param score The \e %CationPiInteractionScore instance to copy.
    # 
    def __init__(score: CationPiInteractionScore) -> None: pass

    ##
    # \brief Constructs a <tt>CationPiInteractionScore</tt> functor with the specified constraints.
    # 
    # \param aro_cat <tt>True</tt> if the first feature argument represents the aromatic- and the second one the cationic-feature, and <tt>False</tt> otherwise.
    # \param min_dist The minimum allowed distance between the cationic- and aromatic-feature centers.
    # \param max_dist The maximum allowed distance between the cationic- and aromatic-feature centers.
    # \param max_ang The maximum allowed angle between the cationic- and aromatic-feature plane normal.
    # 
    def __init__(aro_cat: bool, min_dist: float = 3.5, max_dist: float = 5.5, max_ang: float = 30.0) -> None: pass

    ##
    # \brief 
    # \param func 
    #
    def setDistanceScoringFunction(func: DoubleDoubleFunctor) -> None: pass

    ##
    # \brief 
    # \param func 
    #
    def setAngleScoringFunction(func: DoubleDoubleFunctor) -> None: pass

    ##
    # \brief 
    # \return 
    #
    def getMinDistance() -> float: pass

    ##
    # \brief 
    # \return 
    #
    def getMaxDistance() -> float: pass

    ##
    # \brief 
    # \return 
    #
    def getMaxAngle() -> float: pass

    ##
    # \brief Replaces the current state of \a self with a copy of the state of the \c %CationPiInteractionScore instance \a constr.
    # \param constr The \c %CationPiInteractionScore instance to copy.
    # \return \a self
    # 
    def assign(constr: CationPiInteractionScore) -> CationPiInteractionScore: pass

    minDistance = property(getMinDistance)

    maxDistance = property(getMaxDistance)

    maxAngle = property(getMaxAngle)
