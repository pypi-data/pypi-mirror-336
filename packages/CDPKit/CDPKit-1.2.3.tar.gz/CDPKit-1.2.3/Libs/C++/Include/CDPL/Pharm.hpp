/* 
 * Pharm.hpp 
 *
 * This file is part of the Chemical Data Processing Toolkit
 *
 * Copyright (C) 2003 Thomas Seidel <thomas.seidel@univie.ac.at>
 *
 * This library is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 2 of the License, or (at your option) any later version.
 *
 * This library is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this library; see the file COPYING. If not, write to
 * the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
 * Boston, MA 02111-1307, USA.
 */

/**
 * \file
 * \brief A convenience header including everything that is defined in namespace CDPL::Pharm.
 */

#ifndef CDPL_PHARM_HPP
#define CDPL_PHARM_HPP

#include "CDPL/Pharm/Feature.hpp"
#include "CDPL/Pharm/BasicFeature.hpp"
#include "CDPL/Pharm/FeatureContainer.hpp"
#include "CDPL/Pharm/FeatureSet.hpp"
#include "CDPL/Pharm/Pharmacophore.hpp"
#include "CDPL/Pharm/BasicPharmacophore.hpp"
#include "CDPL/Pharm/FeatureMapping.hpp"
#include "CDPL/Pharm/FeatureTypeHistogram.hpp"
#include "CDPL/Pharm/Feature3DCoordinatesFunction.hpp"
#include "CDPL/Pharm/FeatureContainerProperty.hpp"
#include "CDPL/Pharm/FeatureProperty.hpp"
#include "CDPL/Pharm/FeatureContainerPropertyDefault.hpp"
#include "CDPL/Pharm/FeaturePropertyDefault.hpp"
#include "CDPL/Pharm/ControlParameter.hpp"
#include "CDPL/Pharm/ControlParameterDefault.hpp"
#include "CDPL/Pharm/DataFormat.hpp"
#include "CDPL/Pharm/FeatureType.hpp"
#include "CDPL/Pharm/FeatureGeometry.hpp"
#include "CDPL/Pharm/FeatureGenerator.hpp"
#include "CDPL/Pharm/PatternBasedFeatureGenerator.hpp"
#include "CDPL/Pharm/AromaticFeatureGenerator.hpp"
#include "CDPL/Pharm/HydrophobicFeatureGenerator.hpp"
#include "CDPL/Pharm/HydrophobicAtomFeatureGenerator.hpp"
#include "CDPL/Pharm/HBondAcceptorFeatureGenerator.hpp"
#include "CDPL/Pharm/HBondDonorFeatureGenerator.hpp"
#include "CDPL/Pharm/PosIonizableFeatureGenerator.hpp"
#include "CDPL/Pharm/NegIonizableFeatureGenerator.hpp"
#include "CDPL/Pharm/XBondAcceptorFeatureGenerator.hpp"
#include "CDPL/Pharm/XBondDonorFeatureGenerator.hpp"
#include "CDPL/Pharm/PharmacophoreGenerator.hpp"
#include "CDPL/Pharm/DefaultPharmacophoreGenerator.hpp"
#include "CDPL/Pharm/InteractionAnalyzer.hpp"
#include "CDPL/Pharm/DefaultInteractionAnalyzer.hpp"
#include "CDPL/Pharm/InteractionConstraintConnector.hpp"
#include "CDPL/Pharm/FeatureDistanceConstraint.hpp"
#include "CDPL/Pharm/HydrophobicInteractionConstraint.hpp"
#include "CDPL/Pharm/IonicInteractionConstraint.hpp"
#include "CDPL/Pharm/OrthogonalPiPiInteractionConstraint.hpp"
#include "CDPL/Pharm/ParallelPiPiInteractionConstraint.hpp"
#include "CDPL/Pharm/CationPiInteractionConstraint.hpp"
#include "CDPL/Pharm/HBondingInteractionConstraint.hpp"
#include "CDPL/Pharm/XBondingInteractionConstraint.hpp"
#include "CDPL/Pharm/InteractionPharmacophoreGenerator.hpp"
#include "CDPL/Pharm/FeatureInteractionScore.hpp"
#include "CDPL/Pharm/FeatureInteractionScoreCombiner.hpp"
#include "CDPL/Pharm/FeatureDistanceScore.hpp"
#include "CDPL/Pharm/HydrophobicInteractionScore.hpp"
#include "CDPL/Pharm/IonicInteractionScore.hpp"
#include "CDPL/Pharm/HBondingInteractionScore.hpp"
#include "CDPL/Pharm/XBondingInteractionScore.hpp"
#include "CDPL/Pharm/OrthogonalPiPiInteractionScore.hpp"
#include "CDPL/Pharm/ParallelPiPiInteractionScore.hpp"
#include "CDPL/Pharm/CationPiInteractionScore.hpp"
#include "CDPL/Pharm/SpatialFeatureMapping.hpp"
#include "CDPL/Pharm/PharmacophoreAlignment.hpp"
#include "CDPL/Pharm/FeatureTypeMatchFunctor.hpp"
#include "CDPL/Pharm/FeaturePairDistanceMatchFunctor.hpp"
#include "CDPL/Pharm/FeaturePositionMatchFunctor.hpp"
#include "CDPL/Pharm/FeatureGeometryMatchFunctor.hpp"
#include "CDPL/Pharm/PharmacophoreFitScore.hpp"
#include "CDPL/Pharm/CDFPharmacophoreInputHandler.hpp"
#include "CDPL/Pharm/PMLPharmacophoreInputHandler.hpp"
#include "CDPL/Pharm/CDFFeatureContainerOutputHandler.hpp"
#include "CDPL/Pharm/PMLFeatureContainerOutputHandler.hpp"
#include "CDPL/Pharm/CDFPharmacophoreReader.hpp"
#include "CDPL/Pharm/PMLPharmacophoreReader.hpp"
#include "CDPL/Pharm/CDFFeatureContainerWriter.hpp"
#include "CDPL/Pharm/PMLFeatureContainerWriter.hpp"
#include "CDPL/Pharm/ScreeningDBCreator.hpp"
#include "CDPL/Pharm/ScreeningDBAccessor.hpp"
#include "CDPL/Pharm/ScreeningProcessor.hpp"
#include "CDPL/Pharm/PharmacophoreFitScreeningScore.hpp"
#include "CDPL/Pharm/MoleculeFunctions.hpp"
#include "CDPL/Pharm/PharmacophoreFunctions.hpp"
#include "CDPL/Pharm/FeatureFunctions.hpp"
#include "CDPL/Pharm/FeatureContainerFunctions.hpp"
#include "CDPL/Pharm/FeatureSetFunctions.hpp"
#include "CDPL/Pharm/ControlParameterFunctions.hpp"
#include "CDPL/Pharm/UtilityFunctions.hpp"
#include "CDPL/Pharm/FileScreeningHitCollector.hpp"
#include "CDPL/Pharm/CDFGZPharmacophoreInputHandler.hpp"
#include "CDPL/Pharm/CDFBZ2PharmacophoreInputHandler.hpp"
#include "CDPL/Pharm/CDFGZFeatureContainerOutputHandler.hpp"
#include "CDPL/Pharm/CDFBZ2FeatureContainerOutputHandler.hpp"
#include "CDPL/Pharm/CDFGZPharmacophoreReader.hpp"
#include "CDPL/Pharm/CDFBZ2PharmacophoreReader.hpp"
#include "CDPL/Pharm/CDFGZFeatureContainerWriter.hpp"
#include "CDPL/Pharm/CDFBZ2FeatureContainerWriter.hpp"
#include "CDPL/Pharm/PharmacophoreReader.hpp"
#include "CDPL/Pharm/FeatureContainerWriter.hpp"
# include "CDPL/Pharm/PSDPharmacophoreInputHandler.hpp"
# include "CDPL/Pharm/PSDMoleculeInputHandler.hpp"
# include "CDPL/Pharm/PSDMolecularGraphOutputHandler.hpp"
# include "CDPL/Pharm/PSDPharmacophoreReader.hpp"
# include "CDPL/Pharm/PSDMoleculeReader.hpp"
# include "CDPL/Pharm/PSDMolecularGraphWriter.hpp"
# include "CDPL/Pharm/PSDScreeningDBCreator.hpp"
# include "CDPL/Pharm/PSDScreeningDBAccessor.hpp"

#endif // CDPL_PHARM_HPP
