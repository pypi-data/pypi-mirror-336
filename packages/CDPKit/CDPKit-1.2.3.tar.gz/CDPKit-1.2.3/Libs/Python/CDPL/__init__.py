##
# __init__.py  
#
# This file is part of the Chemical Data Processing Toolkit
#
# Copyright (C) 2003 Thomas Seidel <thomas.seidel@univie.ac.at>
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU Lesser General Public
# License as published by the Free Software Foundation; either
# version 2 of the License, or (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
# Lesser General Public License for more details.
#
# You should have received a copy of the GNU Lesser General Public License
# along with this program; see the file COPYING. If not, write to
# the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
# Boston, MA 02111-1307, USA.
##


from __future__ import absolute_import

import os
import sys
import platform

if sys.version_info[1] >= 8 and platform.system() == 'Windows':
    if 'site-packages' in __file__:
        os.add_dll_directory(os.path.dirname(__file__))
    else:
        os.add_dll_directory(os.path.join(os.path.dirname(os.path.dirname(os.path.dirname(__file__))), '@CDPKIT_EXECUTABLE_INSTALL_DIR@'))

from ._cdpl import *

__version__ = VERSION_STRING
