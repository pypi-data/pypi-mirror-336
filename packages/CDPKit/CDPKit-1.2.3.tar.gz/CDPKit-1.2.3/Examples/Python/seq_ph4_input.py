#!/bin/env python

##
# seq_ph4_input.py 
#
# This file is part of the Pharmical Data Processing Toolkit
#
# Copyright (C) 2003 Thomas Seidel <thomas.seidel@univie.ac.at>
#
# Permission to use, copy, modify, and/or distribute this software for any purpose with
# or without fee is hereby granted.
# THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH REGARD
# TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS. IN
# NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL
# DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER
# IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN
# CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
##


import sys

import CDPL.Pharm as Pharm


# function called for each read pharmacophore
def procPharmacophore(ph4: Pharm.Pharmacophore) -> None: 
    print('Read pharmacophore \'%s\' comprising %s features' % (Pharm.getName(ph4), str(ph4.numFeatures)))

def main() -> None:
    if len(sys.argv) < 2:
        sys.exit('Usage: %s <input pharm. file>' % sys.argv[0])

    # create reader for input pharmacophores (format specified by file extension)
    reader = Pharm.PharmacophoreReader(sys.argv[1]) 
 
    # create an instance of the default implementation of the Pharm.Pharmacophore interface
    ph4 = Pharm.BasicPharmacophore()
    
    # read and process pharmacophores one after the other until the end of input has been reached
    try:
        while reader.read(ph4): 
            try:
                procPharmacophore(ph4)
            except Exception as e:
                sys.exit('Error: processing of pharmacophore failed: ' + str(e))
                
    except Exception as e: # handle exception raised in case of severe read errors
        sys.exit('Error: reading pharmacophore failed: ' + str(e))

    sys.exit(0)
        
if __name__ == '__main__':
    main()
