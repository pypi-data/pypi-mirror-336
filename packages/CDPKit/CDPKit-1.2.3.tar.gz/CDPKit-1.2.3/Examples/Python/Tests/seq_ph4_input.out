Read pharmacophore '1dwc_MIT_1_H_1__D___' comprising 20 features
